packages/connect-ui/src/index.tsx

```diff
@@ -38,29 +61,80 @@ export const ConnectUI = ({ postMessage, clearLegacyView }: ConnectUIProps) => {
 
     useEffect(() => initAnalytics(), []);
 
-    const getComponent = () => {
-        if (!view?.type) return null;
-
-        switch (view.type) {
-            case UI.TRANSPORT:
-                return <Transport />;
-            case UI.REQUEST_PASSPHRASE:
-                return <Passphrase {...view} postMessage={postMessage} />;
-            case 'error':
-                return <ErrorView {...view} />;
-            default:
-                throw new Error(`no such view exists: ${view.type}`);
+    const [Component, Notifications] = useMemo(() => {
+        let component: React.ReactNode | null;
+
+        // component (main screen)
+
+        if (!messages.length) {
+            component = <Loader />;
+        } else {
+            switch (messages[0]?.type) {
+                case 'popup-handshake':
+                    component = <Loader />;
+                    break;
+                case UI.TRANSPORT:
+                    component = <Transport />;
+                    break;
+                case UI.REQUEST_PASSPHRASE:
+                    component = <Passphrase {...messages[0]} postMessage={postMessage} />;
+                    break;
+                case 'error':
+                    component = <ErrorView {...messages[0]} />;
+                    break;
+                default:
+            }
         }
-    };
+
+        // notifications
+        const notifications: React.ReactNode[] = [];
+        messages.forEach(message => {
+            if (message?.type === 'popup-handshake' && message?.payload?.transport?.outdated) {
+                notifications.push(<BridgeUpdateNotification key="bridge-outdated" />);
+            } else if (message?.type === UI_REQUEST.FIRMWARE_OUTDATED) {
+                notifications.push(<FirmwareUpdateNotification key={message.type} />);
+            } else if (message?.type === UI_REQUEST.DEVICE_NEEDS_BACKUP) {
+                notifications.push(<BackupNotification key={message.type} />);
+            }
+            return notifications;
+        });
+
+        return [component, notifications];
+    }, [messages, postMessage]);
+
+    useEffect(() => {
+        if (Component) {
+            clearLegacyView();
+        }
+    }, [Component, clearLegacyView]);
 
     return (
         <ErrorBoundary>
             <GlobalStyle />
             <ThemeWrapper>
                 {/* todo: load translations from somewhere and pass them to intl */}
                 <IntlWrapper>
-                    {getComponent()}
-                    <BottomRightFloatingBar />
+                    <Layout>
+                        <InfoPanel
+                            method={flowInfo?.method}
+                            origin={flowInfo?.settings?.origin}
+                            topSlot={Notifications}
```

Comment:
nit: i know there are no designs but perhaps putting the notifications in the bottom makes more sense? At least the logo stays in the same place :D

Reply:
this is doable but: 
- what about mobile? are  they up, or bottom?
- does it make sense to change it now if there is not no design? this is nothing new, it has been like this for ages.

Reply:
Sure, that's why I marked it with `nit` :) It's just a simple fix that imo makes it look a bit better. Not saying it's required.

Reply:
ok, please lets ignore it. there will be room for this once we get final designs